### ISSUE TITLE
Installing issues

### ISSUE THREAD

**Issue by salvafern** [EMODnet/Btrait#1]
This issue records the error messages when trying to install `Btrait` from github

## Linux
``` r
sessionInfo()
#> R version 3.6.3 (2020-02-29)
#> Platform: x86_64-pc-linux-gnu (64-bit)
#> Running under: Ubuntu 16.04.7 LTS
#> 
#> Matrix products: default
#> BLAS:   /usr/lib/libblas/libblas.so.3.6.0
#> LAPACK: /usr/lib/lapack/liblapack.so.3.6.0
#> 
#> locale:
#>   [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#> [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#> [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#> 
#> attached base packages:
#>   [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#>   [1] reprex_2.0.1   devtools_2.4.5 usethis_2.1.6 
#> 
#> loaded via a namespace (and not attached):
#>   [1] Rcpp_1.0.10       pillar_1.8.1      compiler_3.6.3    later_1.3.0       urlchecker_1.0.1  prettyunits_1.1.1
#> [7] profvis_0.3.7     remotes_2.4.2     tools_3.6.3       digest_0.6.31     pkgbuild_1.4.0    pkgload_1.3.2    
#> [13] tibble_3.1.8      evaluate_0.20     memoise_2.0.1     lifecycle_1.0.3   pkgconfig_2.0.3   rlang_1.0.6      
#> [19] shiny_1.7.4       cli_3.6.0         rstudioapi_0.14   yaml_2.3.7        curl_5.0.0        xfun_0.37        
#> [25] fastmap_1.1.0     knitr_1.42        withr_2.5.0       stringr_1.5.0     desc_1.4.2        fs_1.6.1         
#> [31] vctrs_0.5.2       htmlwidgets_1.6.1 rprojroot_2.0.3   glue_1.6.2        R6_2.5.1          processx_3.8.0   
#> [37] fansi_1.0.4       rmarkdown_2.20    sessioninfo_1.2.2 callr_3.7.3       purrr_1.0.1       magrittr_2.0.3   
#> [43] ps_1.7.2          promises_1.2.0.1  ellipsis_0.3.2    htmltools_0.5.4   mime_0.12         xtable_1.8-4     
#> [49] httpuv_1.6.9      utf8_1.2.3        stringi_1.7.12    miniUI_0.1.1.1    cachem_1.0.6      crayon_1.5.2     

require(devtools)
#> Loading required package: devtools
#> Loading required package: usethis
install_github("EMODnet/Btrait", build_vignettes = TRUE)
#> Downloading GitHub repo EMODnet/Btrait@HEAD
#> 
#> ── R CMD build ─────────────────────────────────────────────────────────────────
#> * checking for file ‘/tmp/Rtmpr3yxXS/remotes559921a248e6/EMODnet-Btrait-e2e1f38/DESCRIPTION’ ... OK
#> * preparing ‘Btrait’:
#> * checking DESCRIPTION meta-information ... OK
#> * cleaning src
#> * installing the package to build vignettes
#> * creating vignettes ... ERROR
#> --- re-building ‘Btrait.Rmd’ using rmarkdown
#> Quitting from lines 89-92 (Btrait.Rmd) 
#> Error: processing vignette 'Btrait.Rmd' failed with diagnostics:
#> could not find function "deparse1"
#> --- failed re-building ‘Btrait.Rmd’
#> 
#> --- re-building ‘BtraitData.Rmd’ using rmarkdown
#> Quitting from lines 170-211 (BtraitData.Rmd) 
#> Error: processing vignette 'BtraitData.Rmd' failed with diagnostics:
#> could not find function "deparse1"
#> --- failed re-building ‘BtraitData.Rmd’
#> 
#> SUMMARY: processing the following files failed:
#>   ‘Btrait.Rmd’ ‘BtraitData.Rmd’
#> 
#> Error: Vignette re-building failed.
#> Execution halted
#> Error: Failed to install 'Btrait' from GitHub:
#>   ! System command 'R' failed
install_github("EMODnet/Btrait", build_vignettes = FALSE)
#> Downloading GitHub repo EMODnet/Btrait@HEAD
#> 
#> ── R CMD build ─────────────────────────────────────────────────────────────────
#> * checking for file ‘/tmp/Rtmpr3yxXS/remotes55992ae1dd69/EMODnet-Btrait-e2e1f38/DESCRIPTION’ ... OK
#> * preparing ‘Btrait’:
#> * checking DESCRIPTION meta-information ... OK
#> * cleaning src
#> * checking vignette meta-information ... ERROR
#> Output(s) listed in ‘build/vignette.rds’ but not in package:
#>   ‘inst/doc/Btrait.pdf’
#>   ‘inst/doc/BtraitData.pdf’
#> Run R CMD build without --no-build-vignettes to re-create
#> Error: Failed to install 'Btrait' from GitHub:
#>   ! System command 'R' failed
```

<sup>Created on 2023-02-16 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

## Windows
Some issues with latex, need to debug further. 

```r
sessionInfo()
#> R version 4.2.0 (2022-04-22 ucrt)
#> Platform: x86_64-w64-mingw32/x64 (64-bit)
#> Running under: Windows 10 x64 (build 19044)
#> 
#> Matrix products: default
#> 
#> locale:
#>   [1] LC_COLLATE=English_United States.utf8  LC_CTYPE=English_United States.utf8   
#> [3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
#> [5] LC_TIME=English_United States.utf8    
#> 
#> attached base packages:
#>   [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#>   [1] reprex_2.0.2   devtools_2.4.3 usethis_2.1.6 
#> 
#> loaded via a namespace (and not attached):
#>   [1] pillar_1.7.0      highr_0.9         compiler_4.2.0    prettyunits_1.1.1
#> [5] remotes_2.4.2     tools_4.2.0       testthat_3.1.5    digest_0.6.29    
#> [9] pkgbuild_1.3.1    pkgload_1.2.4     tibble_3.1.7      memoise_2.0.1    
#> [13] evaluate_0.15     lifecycle_1.0.3   pkgconfig_2.0.3   rlang_1.0.2      
#> [17] cli_3.4.1         rstudioapi_0.13   curl_4.3.2        yaml_2.3.5       
#> [21] xfun_0.31         fastmap_1.1.0     withr_2.5.0       knitr_1.39       
#> [25] vctrs_0.4.1       desc_1.4.1        fs_1.5.2          rprojroot_2.0.3  
#> [29] glue_1.6.2        R6_2.5.1          processx_3.8.0    fansi_1.0.3      
#> [33] rmarkdown_2.14    sessioninfo_1.2.2 callr_3.7.3       purrr_0.3.4      
#> [37] clipr_0.8.0       magrittr_2.0.3    ps_1.7.0          ellipsis_0.3.2   
#> [41] htmltools_0.5.2   utf8_1.2.2        cachem_1.0.6      crayon_1.5.2     
#> [45] brio_1.1.3       

require(devtools)
install_github("EMODnet/Btrait", build_vignettes = TRUE)
#> ✔  checking for file 'C:\Users\salvadorf\AppData\Local\Temp\RtmpQnTVhj\remotes59c37152ba7\EMODnet-Btrait-e2e1f38/DESCRIPTION' (530ms)
#> ─  preparing 'Btrait':
#>   ✔  checking DESCRIPTION meta-information ... 
#> ─  cleaning src
#> ─  installing the package to build vignettes
#> E  creating vignettes (1m 4.8s)
#> --- re-building 'Btrait.Rmd' using rmarkdown
#> Loading required package: Btrait
#> Loading required package: plot3D
#> Warning: package 'plot3D' was built under R version 4.2.2
#> Loading required package: shinydashboard
#> Warning: package 'shinydashboard' was built under R version 4.2.2
#> 
#> Attaching package: 'shinydashboard'
#> 
#> The following object is masked from 'package:graphics':
#>   
#>   box
#> 
#> Warning in citation(package = "Btrait") :
#>   no date field in DESCRIPTION file of package 'Btrait'
#> Warning in system2(..., stdout = if (use_file_stdout()) f1 else FALSE, stderr = f2) :
#>   '"pdflatex"' not found
#> Error: processing vignette 'Btrait.Rmd' failed with diagnostics:
#>   LaTeX failed to compile C:/Users/salvadorf/AppData/Local/Temp/RtmpgRoWXy/Rbuildd781bd5e6d/Btrait/vignettes/Btrait.tex. See https://yihui.org/tinytex/r/#debugging for debugging tips.
#>   --- failed re-building 'Btrait.Rmd'
#> 
#> --- re-building 'BtraitData.Rmd' using rmarkdown
#> Warning in citation(package = "Btrait") :
#>   no date field in DESCRIPTION file of package 'Btrait'
#> Warning in system2(..., stdout = if (use_file_stdout()) f1 else FALSE, stderr = f2) :
#>   '"pdflatex"' not found
#> Error: processing vignette 'BtraitData.Rmd' failed with diagnostics:
#>   LaTeX failed to compile C:/Users/salvadorf/AppData/Local/Temp/RtmpgRoWXy/Rbuildd781bd5e6d/Btrait/vignettes/BtraitData.tex. See https://yihui.org/tinytex/r/#debugging for debugging tips.
#>   --- failed re-building 'BtraitData.Rmd'
#> 
#> SUMMARY: processing the following files failed:
#>   'Btrait.Rmd' 'BtraitData.Rmd'
#> 
#> Error: Vignette re-building failed.
#> Execution halted
#> Error: Failed to install 'Btrait' from GitHub:
#>   ! System command 'Rcmd.exe' failed

install_github("EMODnet/Btrait", build_vignettes = FALSE)
#> ✔  checking for file 'C:\Users\salvadorf\AppData\Local\Temp\RtmpQnTVhj\remotes59cb4319b7\EMODnet-Btrait-e2e1f38/DESCRIPTION' (365ms)
#> ─  preparing 'Btrait':
#>   ✔  checking DESCRIPTION meta-information ... 
#> ─  cleaning src
#> E  checking vignette meta-information ... 
#> Output(s) listed in 'build/vignette.rds' but not in package:
#>   'inst/doc/Btrait.pdf'
#> 'inst/doc/BtraitData.pdf'
#> Run R CMD build without --no-build-vignettes to re-create
#> Error: Failed to install 'Btrait' from GitHub:
#>   ! System command 'Rcmd.exe' failed
```

**Comment by salvafern** [comment]
Added a fix in 55447076 that solves the issues when not building vignettes. 

With vignettes still same issue

```
#> * creating vignettes ... ERROR
#> --- re-building ‘Btrait.Rmd’ using rmarkdown
#> Quitting from lines 89-92 (Btrait.Rmd) 
#> Error: processing vignette 'Btrait.Rmd' failed with diagnostics:
#> could not find function "deparse1"
#> --- failed re-building ‘Btrait.Rmd’
#> 
#> --- re-building ‘BtraitData.Rmd’ using rmarkdown
#> Quitting from lines 170-211 (BtraitData.Rmd) 
#> Error: processing vignette 'BtraitData.Rmd' failed with diagnostics:
#> could not find function "deparse1"
#> --- failed re-building ‘BtraitData.Rmd’
```

**Comment by salvafern** [comment]
There is a dependency on R >= 4, added in 28ee766e29eb2814e03927c8e989a4658a978d65

**Comment by karlines** [comment]
Should be R >= 4.0.0, to avoid deparse1 not being known, as this was introduced in 4.0.0. The alternative is to remove/circumvent this function altogether.

**Comment by salvafern** [comment]
Fixed via: 55447076d6d968eab9d92c9fbea60f7a221d80c7
